File: admin-back/src/main/java/com/asdf/adminback/dto/CertificateSigningDTO.java
package com.asdf.adminback.dto;

public class CertificateSigningDTO {

    private CertificateDataDTO certificateDataDTO;
    private KeyUsageDTO keyUsageDTO;
    private ExtendedKeyUsageDTO extendedKeyUsageDTO;
    private boolean ca;

    public CertificateSigningDTO() { }

    public CertificateDataDTO getCertificateDataDTO() {
        return certificateDataDTO;
    }

    public void setCertificateDataDTO(CertificateDataDTO certificateDataDTO) {
        this.certificateDataDTO = certificateDataDTO;
    }

    public KeyUsageDTO getKeyUsageDTO() {
        return keyUsageDTO;
    }

    public void setKeyUsageDTO(KeyUsageDTO keyUsageDTO) {
        this.keyUsageDTO = keyUsageDTO;
    }

    public ExtendedKeyUsageDTO getExtendedKeyUsageDTO() {
        return extendedKeyUsageDTO;
    }

    public void setExtendedKeyUsageDTO(ExtendedKeyUsageDTO extendedKeyUsageDTO) {
        this.extendedKeyUsageDTO = extendedKeyUsageDTO;
    }

    public boolean isCa() {
        return ca;
    }

    public void setCa(boolean ca) {
        this.ca = ca;
    }
}

File: admin-front/admin-app/src/app/models/CertificateSigningDTO.ts
import { CertificateDataDTO } from "./CertificateDataDTO";
import { ExtendedKeyUsageDTO } from "./ExtendedKeyUsageDTO";
import { KeyUsageDTO } from "./KeyUsageDTO";

export interface CertificateSigningDTO {
    certificateDataDTO: CertificateDataDTO | null;
    keyUsageDTO: KeyUsageDTO | null
    extendedKeyUsageDTO: ExtendedKeyUsageDTO | null;
    ca: boolean | null;
}